src/action.rs

```diff
@@ -75,17 +71,55 @@ pub enum Action {
     PeerTryWrite(PeerTryWriteAction),
     PeerTryRead(PeerTryReadAction),
 
-    PeerHandshakingInit(PeerHandshakingInitAction),
-
-    PeerConnectionMessageWriteInit(PeerConnectionMessageWriteInitAction),
-    PeerConnectionMessagePartWritten(PeerConnectionMessagePartWrittenAction),
-    PeerConnectionMessageWriteError(PeerConnectionMessageWriteErrorAction),
-    PeerConnectionMessageWriteSuccess(PeerConnectionMessageWriteSuccessAction),
+    // chunk read
+    PeerChunkReadInit(PeerChunkReadInitAction),
+    PeerChunkReadPart(PeerChunkReadPartAction),
+    PeerChunkReadDecrypt(PeerChunkReadDecryptAction),
+    PeerChunkReadReady(PeerChunkReadReadyAction),
+    PeerChunkReadError(PeerChunkReadErrorAction),
+
+    // chunk write
+    PeerChunkWriteSetContent(PeerChunkWriteSetContentAction),
+    PeerChunkWriteEncryptContent(PeerChunkWriteEncryptContentAction),
+    PeerChunkWriteCreateChunk(PeerChunkWriteCreateChunkAction),
+    PeerChunkWritePart(PeerChunkWritePartAction),
+    PeerChunkWriteReady(PeerChunkWriteReadyAction),
+    PeerChunkWriteError(PeerChunkWriteErrorAction),
+
+    // binary message read
+    PeerBinaryMessageReadInit(PeerBinaryMessageReadInitAction),
+    PeerBinaryMessageReadChunkReady(PeerBinaryMessageReadChunkReadyAction),
+    PeerBinaryMessageReadSizeReady(PeerBinaryMessageReadSizeReadyAction),
+    PeerBinaryMessageReadReady(PeerBinaryMessageReadReadyAction),
+    PeerBinaryMessageReadError(PeerBinaryMessageReadErrorAction),
+
+    // binary message write
+    PeerBinaryMessageWriteSetContent(PeerBinaryMessageWriteSetContentAction),
+    PeerBinaryMessageWriteNextChunk(PeerBinaryMessageWriteNextChunkAction),
+    PeerBinaryMessageWriteReady(PeerBinaryMessageWriteReadyAction),
+    PeerBinaryMessageWriteError(PeerBinaryMessageWriteErrorAction),
 
-    PeerConnectionMessageReadInit(PeerConnectionMessageReadInitAction),
-    PeerConnectionMessagePartRead(PeerConnectionMessagePartReadAction),
-    PeerConnectionMessageReadError(PeerConnectionMessageReadErrorAction),
-    PeerConnectionMessageReadSuccess(PeerConnectionMessageReadSuccessAction),
+    PeerHandshakingInit(PeerHandshakingInitAction),
+    PeerHandshakingConnectionInitialize(PeerHandshakingConnectionInitializeAction),
```

Comment:
Maybe lets stay consistent and use same naming. So `Initialize` -> `Init`. Same for other actions or states.

Reply:
Also it's not very clear what this action does. It's better to use longer names than make it ambigous.

So it would be better if we include "message" in there. So something like: `PeerHandshakingConnectionMessageInit`.